File: nim/reverse-string/reverse_string.nim

import unicode

iterator reverseRunes(input: string): Rune =
  let runes = input.toRunes()
  for i in countdown(runes.len, 1):
    yield runes[i - 1]

func reverse*(input: string): string =
  for rune in input.reverseRunes:
    result.add(rune)

File: csharp/reverse-string/ReverseString.cs
using System;

public static class ReverseString
{
    public static string Reverse(string input)
    {
        var reversedChars = new char[input.Length];
        
        for (int i = 0; i < input.Length; i++)
            reversedChars[i] = input[input.Length - 1 - i];

        return new string(reversedChars);
    }
}
